### PR TITLE
dev/core#4421 Fix some attempts to count NULL (php8.x)

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
See https://lab.civicrm.org/dev/core/-/issues/4421

Before
----------------------------------------
As @larssandergreen diagnosed we have a few instances of htis pattern of counting a possibly NULL value

After
----------------------------------------
switch to checking if it is not empty

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Mailing/Form/Unsubscribe.php [hunk 1]
```diff
@@ -78,9 +78,10 @@ public function preProcess() {
     $this->_email = $email;
 
     $groups = CRM_Mailing_Event_BAO_MailingEventUnsubscribe::unsub_from_mailing($job_id, $queue_id, $hash, TRUE);
-    $this->assign('groups', $groups);
+    $this->assign('groups', $groups ?? []);
```
Comment (eileenmcnaughton) [inline CRM/Mailing/Form/Unsubscribe.php:81]: This cast to array is more about consistency - I don't think it tries to process it if empty due the nasty groupExist pattern

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
test this please

File: CRM/Mailing/Form/Unsubscribe.php [hunk 2]
```diff
@@ -119,7 +121,7 @@ public function postProcess() {
     // Email address verified
     $groups = CRM_Mailing_Event_BAO_MailingEventUnsubscribe::unsub_from_mailing($this->_job_id, $this->_queue_id, $this->_hash);
 
-    if (count($groups)) {
+    if (!empty($groups)) {
```
Comment (colemanw) [inline CRM/Mailing/Form/Unsubscribe.php:124]: This is fine, but FYI for a variable that has definitely been defined (as this one was 1 line above) you don't need count or empty, the if() will cast it to boolean just fine.

**Pull request merged**